File: CPP/119_PascalsTriangleSequel.cpp
// 119 - Pascal's Triangle 2

vector<int> getRow(int rowIndex) {
    vector<vector<int>> p;
    for(int i = 0; i <= rowIndex; i++) {
        vector<int> row;
        row.resize(i + 1, 1);
        for(int j = 1; j < i; j++) {
            row[j] = p[i - 1][j - 1] + p[i - 1][j];
        }
        p.push_back(row);
    }
    return p[rowIndex];
}

File: Java/119_PascalsTriangleSequel.java
// 119 - Pascal's Triangle 2

public List<Integer> getRow(int rowIndex) {
    List<List<Integer>> p = new ArrayList<>(rowIndex);

    for(int i = 0; i <= rowIndex; i++) {
        List<Integer> row = new ArrayList<>(Collections.nCopies(i + 1, 1));
        for(int j = 1; j < i; j++) {
            row.set(j, p.get(i - 1).get(j - 1) + p.get(i - 1).get(j));
        }
        p.add(row);
    }

    return p.get(rowIndex);
}